### PR TITLE
feat(speculative): add tqdm training progress bar to EAGLE and DFlash recipes

### PR DESCRIPTION
**Pull request by khazic**
## What

The speculative-decoding training loops (EAGLE-1/2, EAGLE-3 / P-EAGLE, and DFlash) only emit a log line every `log_every_steps` optimizer steps. There is no view of overall progress: no step/total, no percentage, and no time estimate, so a long run looks stalled between log lines.

This PR adds a rank-0 tqdm progress bar to every speculative recipe, showing `step/total`, percentage, elapsed time, ETA, and seconds per step, with loss/accuracy/lr attached as the bar postfix at each logging interval.

## How

- Generalize `BaseRecipe._make_progress_bar` to accept explicit `total`/`initial` arguments. These recipes hand-roll their training loop and have no `step_scheduler`; they pass `total_optim_steps` and the resumed `runtime.global_step` instead. Without arguments the helper behaves exactly as before (reads `step_scheduler`), so the existing `train_ft`/`kd`/`seq_cls`/VLM/retrieval callers are unchanged.
- `train_eagle1.py`, `train_eagle3.py`, and `train_dflash.py`: create the bar at loop start, advance it once per optimizer step (including the trailing grad-accum flush step), set the postfix at each `log_every_steps` interval, and close it on exit (EAGLE-3 closes in its existing `try/finally`).
- Coverage: EAGLE-2 inherits EAGLE-1's loop unchanged; P-EAGLE runs through the EAGLE-3 loop. So EAGLE-1/2/3, P-EAGLE, and DFlash are all covered.
- Periodic `logger.info` lines and W&B logging are unchanged.

## Tests

- `TestMakeProgressBar::test_explicit_total_and_initial_skip_step_scheduler`: explicit args bypass `step_scheduler` entirely.
- `test_progress_bar_advances_per_optim_step` (eagle1, eagle3, dflash): the bar advances once per optimizer step including the trailing flush, receives a loss/acc/lr postfix, and is closed.
- Bare test-harness recipes now set `total_optim_steps` like the real `setup()` does.

`pytest` over the EAGLE/DFlash recipe suites, `test_base_recipe.py`, and `test_train_ft.py`: 230 passed. ruff clean.

## Notes

`main` currently has two failing `TestFormatLoad` checkpoint unit tests (fixed by #2519 / #2520); if L0 shows those two failures here, they are pre-existing and unrelated to this change.

### CODE REVIEW

File: nemo_automodel/recipes/llm/train_eagle1.py [hunk 1]
```diff
@@ -583,6 +583,7 @@ def run_train_validation_loop(self):
         except TypeError:
             batches_per_epoch = None
         is_ddp = isinstance(self.trainer_module, DistributedDataParallel)
+        pbar = self._make_progress_bar(total=self.total_optim_steps, initial=self.runtime.global_step)
```
[the remark this answers is not in the document]
Comment (khazic) [reply nemo_automodel/recipes/llm/train_eagle1.py:586]: Good catch, fixed in a8072c00b. Wrapped the loop and final-checkpoint calls in try/finally and close the bar in finally, matching EAGLE-3. The same gap existed in DFlash, so I fixed it there too, and added a regression test in each that asserts the bar is closed when the loop raises.

**Review by HuiyingLi — APPROVED** [review]
